backends/bmv2/common/extern.cpp

```diff
@@ -88,13 +88,21 @@ ExternConverter::convertExternObject(ConversionContext* ctxt,
     }
 }
 
+/// This method is invoked for all externs that do not have a registered
+/// conversion, i.e., unknown by the architecture.
 void
-ExternConverter::convertExternInstance(ConversionContext* ,
-                                       const IR::Declaration* ,
+ExternConverter::convertExternInstance(ConversionContext* ctxt,
+                                       const IR::Declaration* decl,
                                        const IR::ExternBlock* eb,
                                        const bool& emitExterns) {
     if (!emitExterns)
         ::error(ErrorType::ERR_UNKNOWN, "extern instance", eb->type->name);
+    auto attr = new Util::JsonArray();
+    // TODO: add attributes.  It is not clear how
+    // JsonObjects::add_extern_attribute can be used to insert attributes,
```

Comment:
The "attributes" here should be the (compile-time) arguments to the extern constructor (as hexadecimal strings). I think this is very different from `IR::Attribute`, which seems to be used for P4_14 "externs" only.

Reply:
OK, I will add the constructor arguments.
I assume that only scalar values are supported.

Reply:
Ideally enum members would be supported as well. We have 2 choices: the P4 name of the enum members or the numerical value allocated by the backend (since the attribute type can be "hexstr" or "string"). My preference is to use the numerical value, and I think that's what @derekso1 did for PSA counters.

Reply:
Yes, I opted for numerical values since as Antonin said, them being "hexstr" makes for easier integration with bmv2. At least in psa_switch, many enums defined in psa.p4 are used to instantiate externs so making them numerical made it easier to create said externs which support only "hexstr" and "string" for attributes.